src/core/ensemble_scheduler.cc

```diff
@@ -41,22 +41,57 @@ namespace {
 
 class EnsembleContext;
 
+using IterationCount = size_t;
```

Comment:
Why is this necessary?

Reply:
For clarity, too many `size_t` for different kinds of variables.